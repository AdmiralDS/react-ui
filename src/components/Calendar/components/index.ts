export * from './Years';
export * from './Button';
export * from './Day';
export * from './DayNames';
export * from './Month';
export * from './Months';
export * from './Panel';
export * from './Week';
