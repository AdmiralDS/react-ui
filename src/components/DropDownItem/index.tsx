import type { HTMLAttributes } from 'react';
import * as React from 'react';
import styled, { css } from 'styled-components';
import { backgroundColor, colorTextMixin, DropDownItemDimension, paddings, styleTextMixin } from './mixins';

export interface DropDownItemProps extends HTMLAttributes<HTMLLIElement> {
  /** Отключение секции */
  disabled?: boolean;
  /** Размер DropDownItems */
  dimension?: DropDownItemDimension;
  /** Активная секция DropDownItems */
  selected?: boolean;
  /** Акцентная секция DropDownItems */
  hovered?: boolean;
  /** Значение DropDownItems */
  value?: string | number | undefined;
}

const hoverStyle = css`
  background-color: ${(p) => p.theme.color.background.secondary};
  cursor: pointer;
`;

const Item = styled.li<{
  dimension: DropDownItemDimension;
  selected?: boolean;
  hovered?: boolean;
  width?: number;
  id?: string;
}>`
  display: flex;
  align-items: center;
  user-select: none;
  flex-flow: wrap;
  position: relative;
  justify-content: space-between;
  outline: none;
  white-space: pre;
  margin: 0;
  ${paddings}
  ${styleTextMixin}
  ${colorTextMixin}
  ${backgroundColor}
  ${(p) => p.hovered && hoverStyle};

  &:hover {
    ${hoverStyle}
  }

  &&[data-disabled='true'] {
    cursor: not-allowed;
    background-color: ${(p) => p.theme.color.background.primary};
    color: ${(p) => p.theme.color.text.tertiary};
    && *[fill^='#'] {
      fill: ${(p) => p.theme.color.text.tertiary};
    }
  }
`;

export const DropDownItem = React.forwardRef<HTMLLIElement, DropDownItemProps>(
  ({ children, disabled, hovered, dimension = 'l', selected = false, id, ...props }, ref) => {
    return (
      <Item
        ref={ref}
        dimension={dimension}
        tabIndex={!disabled ? props.tabIndex || 0 : undefined}
        selected={selected}
        hovered={hovered}
        id={id}
        data-disabled={disabled}
        {...props}
      >
        {children}
      </Item>
    );
  },
);

DropDownItem.displayName = 'DropDownItem';
