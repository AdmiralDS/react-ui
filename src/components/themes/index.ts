export * from './light';
export * from './dark';
export type { Theme } from './common';
