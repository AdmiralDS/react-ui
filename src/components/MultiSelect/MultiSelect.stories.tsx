import React, { Fragment, useState } from 'react';
import styled from 'styled-components';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import { MultiSelect } from './index';
import { withDesign } from 'storybook-addon-designs';
import { DropDownItem } from '#src/components/DropDownItem';
import { CheckboxField } from '#src/components/form/CheckboxField';
import { ReactComponent as CloseOutline } from '@admiral-ds/icons/build/service/CloseOutline.svg';
import { ReactComponent as CardSolid } from '@admiral-ds/icons/build/finance/CardSolid.svg';
import { ReactComponent as StarSolid } from '@admiral-ds/icons/build/system/StarSolid.svg';
import { typography } from '../Typography';

const StyledText = styled.div`
  ${typography['Body/Body 2 Long']}
  color: ${({ theme }) => theme.color.text.primary};
  pointer-events: none;
`;

const Desc = styled.div`
  font-family: 'VTB Group UI';
  font-size: 16px;
  line-height: 24px;
`;
const Description = () => (
  <Desc>Мульти-селекты нужны для выбора нескольких значений из зараннее заданных в списке опций.</Desc>
);

const WrapperItem = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  & > span {
    padding-left: 3px;
  }
`;

export default {
  title: 'Example/ MultiSelect',
  component: MultiSelect,
  decorators: [withDesign],
  parameters: {
    docs: {
      source: {
        type: 'code',
      },
    },
    componentSubtitle: <Description />,
    design: [
      {
        type: 'figma',
        url: 'https://www.figma.com/file/CC0WL5u9TPtZpyLbbAGFGt/Admiral-2.0-UI-Kit?node-id=39%3A72815',
      },
      {
        type: 'figma',
        url: 'https://www.figma.com/file/CC0WL5u9TPtZpyLbbAGFGt/Admiral-2.0-UI-Kit?node-id=39%3A72897',
      },
      {
        type: 'figma',
        url: 'https://www.figma.com/file/CC0WL5u9TPtZpyLbbAGFGt/Admiral-2.0-UI-Kit?node-id=39%3A72973',
      },
    ],
  },
  argTypes: {
    dimension: {
      options: ['xl', 'm', 's'],
      control: { type: 'radio' },
      defaultValue: 'xl',
    },
    disabled: {
      control: { type: 'boolean' },
    },
  },
} as ComponentMeta<typeof MultiSelect>;

const Template: ComponentStory<typeof MultiSelect> = (props) => {
  const [selectValue, setValue] = useState<string[]>([]);

  const options = [
    { id: 1, label: 'Option one', value: '1', disabled: true },
    { id: 2, label: 'Option two', value: '2' },
    { id: 3, label: 'Option three', value: '3' },
    { id: 4, label: 'Option four', value: '4' },
    { id: 5, label: 'Option five', value: '5' },
    { id: 6, label: 'Option six', value: '6' },
    { id: 7, label: 'Option seven', value: '7' },
  ];

  return (
    <MultiSelect {...props} value={selectValue} placeholder="Placeholder" onChange={(value) => setValue(value)}>
      {options.map((item) => (
        <DropDownItem key={item.id} role="option" value={item.label} disabled={item.disabled}>
          <CheckboxField
            dimension={props.dimension === 's' ? 's' : 'm'}
            disabled={item.disabled}
            checked={selectValue.indexOf(item.label) > -1}
            readOnly
          >
            {item.label}
          </CheckboxField>
        </DropDownItem>
      ))}
    </MultiSelect>
  );
};

const Template2: ComponentStory<typeof MultiSelect> = (props) => {
  const [selectValue, setValue] = useState<string[]>([]);
  const iconWidth = props.dimension === 's' ? 20 : 24;
  const options = [
    { id: 1, label: 'Option one', value: '1' },
    { id: 2, label: 'Option two', value: '2' },
    { id: 3, label: 'Option three', value: '3' },
    { id: 4, label: 'Option four', value: '4' },
    { id: 5, label: 'Option five', value: '5' },
    { id: 6, label: 'Option six', value: '6' },
    { id: 7, label: 'Option seven', value: '7' },
  ];

  const handleKey: React.KeyboardEventHandler<HTMLLIElement> = (e) => {
    if (e.keyCode === 32) {
      setValue([]);
    }
  };

  return (
    <MultiSelect {...props} value={selectValue} placeholder="Placeholder" onChange={(value) => setValue(value)}>
      <DropDownItem onClick={() => setValue([])} onKeyDown={handleKey} tabIndex={0}>
        <WrapperItem>
          <CloseOutline width={iconWidth} height={iconWidth} />
          <span> Снять выбор</span>
        </WrapperItem>
      </DropDownItem>
      {options.map((item) => (
        <DropDownItem role="option" value={item.label} key={item.id}>
          <CheckboxField
            onChange={() => ''}
            dimension={props.dimension === 's' ? 's' : 'm'}
            checked={selectValue.indexOf(item.label) > -1}
          >
            {item.label}
          </CheckboxField>
        </DropDownItem>
      ))}
    </MultiSelect>
  );
};

const Template3: ComponentStory<typeof MultiSelect> = (props) => {
  const [selectValue, setValue] = useState<string[]>([]);
  const iconWidth = props.dimension === 's' ? 24 : 30;
  const closeIconWidth = props.dimension === 's' ? 20 : 24;
  const category = [
    {
      name: 'Категория 1',
      id: '1',
      content: [
        {
          id: '2',
          label: 'Option one',
          value: '1',
          icon: <CardSolid width={iconWidth} height={iconWidth} />,
          subText: 'Дополнительный текст',
        },
        {
          id: '3',
          label: 'Option two',
          value: '2',
          icon: <CardSolid width={iconWidth} height={iconWidth} />,
          subText: 'Дополнительный текст',
        },
        {
          id: '4',
          label: 'Option three',
          value: '3',
          icon: <CardSolid width={iconWidth} height={iconWidth} />,
          subText: 'Дополнительный текст',
        },
        {
          id: '5',
          label: 'Option four',
          value: '4',
          icon: <CardSolid width={iconWidth} height={iconWidth} />,
          subText: 'Дополнительный текст',
        },
      ],
    },
    {
      name: 'Категория 2',
      id: '9',
      content: [
        {
          id: '10',
          label: 'Option five',
          value: '5',
          icon: <StarSolid width={iconWidth} height={iconWidth} />,
          subText: 'Дополнительный текст',
        },
        {
          id: '11',
          label: 'Option six',
          value: '6',
          icon: <StarSolid width={iconWidth} height={iconWidth} />,
          subText: 'Дополнительный текст',
        },
        {
          id: '12',
          label: 'Option seven',
          value: '7',
          subText: 'Дополнительный текст',
          icon: <StarSolid width={iconWidth} height={iconWidth} />,
        },
        {
          id: '13',
          label: 'Option eight',
          value: '8',
          icon: <StarSolid width={iconWidth} height={iconWidth} />,
          subText: 'Дополнительный текст',
        },
      ],
    },
  ];

  const handleKey: React.KeyboardEventHandler<HTMLLIElement> = (e) => {
    if (e.keyCode === 32) {
      setValue([]);
    }
  };

  return (
    <MultiSelect {...props} value={selectValue} placeholder="Placeholder" onChange={(value) => setValue(value)}>
      <DropDownItem onClick={() => setValue([])} onKeyDown={handleKey} tabIndex={0}>
        <WrapperItem>
          <CloseOutline width={closeIconWidth} height={closeIconWidth} />
          <span> Снять выбор</span>
        </WrapperItem>
      </DropDownItem>
      {category.map((item, index) => (
        <Fragment key={index}>
          <DropDownItem disabled key={item.id}>
            <StyledText> {item.name}</StyledText>
          </DropDownItem>
          {item.content.map((subCategory) => (
            <DropDownItem tabIndex={0} role="option" key={subCategory.id} value={subCategory.label} id={subCategory.id}>
              <CheckboxField
                dimension={props.dimension === 's' ? 's' : 'm'}
                onChange={() => ''}
                checked={selectValue.indexOf(subCategory.label) > -1}
                extraText={subCategory.subText}
              >
                {subCategory.label}
              </CheckboxField>
              {subCategory.icon}
            </DropDownItem>
          ))}
        </Fragment>
      ))}
    </MultiSelect>
  );
};

const Template4: ComponentStory<typeof MultiSelect> = (props) => {
  const [selectValue, setValue] = useState<string[]>([]);
  const options = [
    { id: 1, label: 'Номер карты ********45', value: '1' },
    { id: 2, label: 'Номер карты ********46', value: '2' },
    { id: 3, label: 'Номер карты ********47', value: '3' },
    { id: 4, label: 'Номер карты ********48', value: '4' },
    { id: 5, label: 'Номер карты ********49', value: '5' },
    { id: 6, label: 'Номер карты ********50', value: '6' },
    { id: 7, label: 'Номер карты ********51', value: '7' },
  ];

  const handleDeclension = (value: number) => (value > 4 ? 'карт' : 'карты');
  const handleKey: React.KeyboardEventHandler<HTMLLIElement> = (e) => {
    if (e.keyCode === 32) {
      setValue([]);
    }
  };

  return (
    <MultiSelect
      {...props}
      renderInputValue={() =>
        selectValue.length > 1 ? (
          <div>
            Выбранно {selectValue.length} {handleDeclension(selectValue.length)}
          </div>
        ) : (
          selectValue.map((item, index) => (
            <div key={index} style={{ display: 'flex', alignItems: 'center' }}>
              <div>{item}</div>
              <CardSolid width={26} />
            </div>
          ))
        )
      }
      value={selectValue}
      placeholder="Placeholder"
      onChange={(value) => setValue(value)}
    >
      <DropDownItem onClick={() => setValue([])} onKeyDown={handleKey} tabIndex={0}>
        <WrapperItem>
          <CloseOutline width={24} height={24} />
          <span> Снять выбор</span>
        </WrapperItem>
      </DropDownItem>
      {options.map((item) => (
        <DropDownItem role="option" value={item.label} key={item.id}>
          <CheckboxField
            dimension={props.dimension === 's' ? 's' : 'm'}
            onChange={() => ''}
            checked={selectValue.indexOf(item.label) > -1}
          >
            {item.label}
          </CheckboxField>
        </DropDownItem>
      ))}
    </MultiSelect>
  );
};

export const MultiSelectDefault = Template.bind({});
MultiSelectDefault.storyName = 'Дефолтный мультиселект';

export const MultiSelectFilter = Template2.bind({});
MultiSelectFilter.storyName = 'Мультиселект с удалением всех опций';

export const MultiSelectSubGroup = Template3.bind({});
MultiSelectSubGroup.storyName = 'Мультиселект с подгруппами';

export const MultiSelectSelectChips = Template4.bind({});
MultiSelectSelectChips.storyName = 'Кастомный мультиселект';
