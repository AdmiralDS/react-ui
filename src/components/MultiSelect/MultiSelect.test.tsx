import { DropDownItem } from '#src/components/DropDownItem';
import { MultiSelect } from '#src/components/MultiSelect';
import { LIGHT_THEME } from '#src/components/themes';
import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import userEvent from '@testing-library/user-event';

describe('MultiSelect', () => {
  beforeEach(() => {
    jest.clearAllTimers();
  });

  it('should render component', () => {
    const wrapper = render(
      <ThemeProvider theme={LIGHT_THEME}>
        <MultiSelect value={[]} onChange={() => ''}>
          <DropDownItem value={1}>Option</DropDownItem>
        </MultiSelect>
      </ThemeProvider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render small components', () => {
    const wrapper = render(
      <ThemeProvider theme={LIGHT_THEME}>
        <MultiSelect dimension={'s'} value={[]} onChange={() => ''}>
          <DropDownItem value={'1'}>Option</DropDownItem>
        </MultiSelect>
      </ThemeProvider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should open dropDown  when user clicks on component', () => {
    render(
      <ThemeProvider theme={LIGHT_THEME}>
        <MultiSelect data-testid="select" value={[]} onChange={() => ''}>
          <DropDownItem value={'1'}>Option one</DropDownItem>
          <DropDownItem value={'2'}>Option two</DropDownItem>
          <DropDownItem value={'3'}>Option three</DropDownItem>
        </MultiSelect>
      </ThemeProvider>,
    );

    fireEvent.click(screen.getByTestId('select'));

    expect(screen.getByText(`Option one`)).toHaveTextContent(`Option one`);
  });

  it('should open dropDown  when user clicks on option one', () => {
    render(
      <ThemeProvider theme={LIGHT_THEME}>
        <MultiSelect data-testid="select" value={[]} onChange={() => ''}>
          <DropDownItem value={'1'}>Option one</DropDownItem>
          <DropDownItem value={'2'}>Option two</DropDownItem>
          <DropDownItem value={'3'}>Option three</DropDownItem>
        </MultiSelect>
      </ThemeProvider>,
    );

    fireEvent.click(screen.getByTestId('select'));

    expect(screen.getByText(`Option one`)).toHaveTextContent(`Option one`);
    fireEvent.click(screen.getByText(`Option one`));
    expect(screen.getByText(`Option one`)).toHaveTextContent(`Option one`);
  });

  it('should hide dropdown when user presses Escape', () => {
    render(
      <ThemeProvider theme={LIGHT_THEME}>
        <MultiSelect data-testid="select" value={[]} onChange={() => ''}>
          <DropDownItem value={'1'}>Option one</DropDownItem>
          <DropDownItem value={'2'}>Option two</DropDownItem>
          <DropDownItem value={'3'}>Option three</DropDownItem>
        </MultiSelect>
      </ThemeProvider>,
    );

    userEvent.tab();
    const select = screen.getByTestId('select');

    fireEvent.keyDown(select, { key: ' ', code: '32' });

    const menu = screen.getAllByRole('listbox')[0];

    fireEvent.keyDown(menu, { key: 'Escape', code: 'Escape' });

    expect(screen.queryByRole('listbox')).not.toBeInTheDocument();
  });
});
