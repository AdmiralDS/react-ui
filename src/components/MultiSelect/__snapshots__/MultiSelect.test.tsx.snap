// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiSelect should render component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c5 {
  -webkit-transition: -webkit-transform 0.3s ease-in-out;
  -webkit-transition: transform 0.3s ease-in-out;
  transition: transform 0.3s ease-in-out;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.c5 *[fill^='#'] {
  fill: #626F84;
}

.c5.c5[data-disabled] {
  pointer-events: none;
  fill: #A7B0BE;
}

.c5:not([data-disabled]):hover {
  cursor: pointer;
}

.c5:not([data-disabled]):hover *[fill^='#'] {
  fill: #0A54C2;
}

.c2 {
  white-space: nowrap;
  position: relative;
  color: #2B313B;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-flex-flow: nowrap;
  -ms-flex-flow: nowrap;
  flex-flow: nowrap;
  border: 0;
  pointer-events: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 16px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  font-family: 'VTB Group UI',sans-serif;
  box-sizing: border-box;
  border-radius: 4px;
  height: inherit;
  cursor: pointer;
}

.c4 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: 10px;
}

.c4 > * {
  display: block;
  width: 24px;
  height: 24px;
}

.c4 > *:not(:first-child) {
  margin-left: 8px;
}

.c1 {
  cursor: pointer;
  background: #FFFFFF;
  position: relative;
  border-radius: 4px;
  height: 38px;
  border-color: #8A96A8;
  border-width: 1px;
  border-style: solid;
}

[data-focus-within] .c0,
.c1:focus {
  border-color: #0949AA;
}

.c1:hover {
  border-color: #0A54C2;
}

.c1:focus {
  outline: none;
}

.c1:active {
  outline: none;
}

[data-status='success'] .c0 {
  border-color: #2ABB5B;
}

[data-status='error'] .c0 {
  border-color: #DA0B20;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  overflow: hidden;
  position: relative;
  height: 32px;
  padding-right: 46px;
  pointer-events: none;
}

<body>
    <div>
      <div
        class="c0 c1"
        tabindex="0"
      >
        <div
          class="c2"
        >
          <div
            class="c3"
          />
          <div
            class="c4"
          >
            <div
              aria-hidden="true"
              class="c5"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .c5 {
  -webkit-transition: -webkit-transform 0.3s ease-in-out;
  -webkit-transition: transform 0.3s ease-in-out;
  transition: transform 0.3s ease-in-out;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.c5 *[fill^='#'] {
  fill: #626F84;
}

.c5.c5[data-disabled] {
  pointer-events: none;
  fill: #A7B0BE;
}

.c5:not([data-disabled]):hover {
  cursor: pointer;
}

.c5:not([data-disabled]):hover *[fill^='#'] {
  fill: #0A54C2;
}

.c2 {
  white-space: nowrap;
  position: relative;
  color: #2B313B;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-flex-flow: nowrap;
  -ms-flex-flow: nowrap;
  flex-flow: nowrap;
  border: 0;
  pointer-events: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 16px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  font-family: 'VTB Group UI',sans-serif;
  box-sizing: border-box;
  border-radius: 4px;
  height: inherit;
  cursor: pointer;
}

.c4 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: 10px;
}

.c4 > * {
  display: block;
  width: 24px;
  height: 24px;
}

.c4 > *:not(:first-child) {
  margin-left: 8px;
}

.c1 {
  cursor: pointer;
  background: #FFFFFF;
  position: relative;
  border-radius: 4px;
  height: 38px;
  border-color: #8A96A8;
  border-width: 1px;
  border-style: solid;
}

[data-focus-within] .c0,
.c1:focus {
  border-color: #0949AA;
}

.c1:hover {
  border-color: #0A54C2;
}

.c1:focus {
  outline: none;
}

.c1:active {
  outline: none;
}

[data-status='success'] .c0 {
  border-color: #2ABB5B;
}

[data-status='error'] .c0 {
  border-color: #DA0B20;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  overflow: hidden;
  position: relative;
  height: 32px;
  padding-right: 46px;
  pointer-events: none;
}

<div>
    <div
      class="c0 c1"
      tabindex="0"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        />
        <div
          class="c4"
        >
          <div
            aria-hidden="true"
            class="c5"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`MultiSelect should render small components 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c5 {
  -webkit-transition: -webkit-transform 0.3s ease-in-out;
  -webkit-transition: transform 0.3s ease-in-out;
  transition: transform 0.3s ease-in-out;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.c5 *[fill^='#'] {
  fill: #626F84;
}

.c5.c5[data-disabled] {
  pointer-events: none;
  fill: #A7B0BE;
}

.c5:not([data-disabled]):hover {
  cursor: pointer;
}

.c5:not([data-disabled]):hover *[fill^='#'] {
  fill: #0A54C2;
}

.c2 {
  white-space: nowrap;
  position: relative;
  color: #2B313B;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-flex-flow: nowrap;
  -ms-flex-flow: nowrap;
  flex-flow: nowrap;
  border: 0;
  pointer-events: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 12px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  font-family: 'VTB Group UI',sans-serif;
  box-sizing: border-box;
  border-radius: 4px;
  height: inherit;
  cursor: pointer;
}

.c4 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: 10px;
}

.c4 > * {
  display: block;
  width: 20px;
  height: 20px;
}

.c4 > *:not(:first-child) {
  margin-left: 8px;
}

.c1 {
  cursor: pointer;
  background: #FFFFFF;
  position: relative;
  border-radius: 4px;
  height: 30px;
  border-color: #8A96A8;
  border-width: 1px;
  border-style: solid;
}

[data-focus-within] .c0,
.c1:focus {
  border-color: #0949AA;
}

.c1:hover {
  border-color: #0A54C2;
}

.c1:focus {
  outline: none;
}

.c1:active {
  outline: none;
}

[data-status='success'] .c0 {
  border-color: #2ABB5B;
}

[data-status='error'] .c0 {
  border-color: #DA0B20;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  overflow: hidden;
  position: relative;
  height: 32px;
  padding-right: 46px;
  pointer-events: none;
}

<body>
    <div>
      <div
        class="c0 c1"
        tabindex="0"
      >
        <div
          class="c2"
        >
          <div
            class="c3"
          />
          <div
            class="c4"
          >
            <div
              aria-hidden="true"
              class="c5"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .c5 {
  -webkit-transition: -webkit-transform 0.3s ease-in-out;
  -webkit-transition: transform 0.3s ease-in-out;
  transition: transform 0.3s ease-in-out;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.c5 *[fill^='#'] {
  fill: #626F84;
}

.c5.c5[data-disabled] {
  pointer-events: none;
  fill: #A7B0BE;
}

.c5:not([data-disabled]):hover {
  cursor: pointer;
}

.c5:not([data-disabled]):hover *[fill^='#'] {
  fill: #0A54C2;
}

.c2 {
  white-space: nowrap;
  position: relative;
  color: #2B313B;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-flex-flow: nowrap;
  -ms-flex-flow: nowrap;
  flex-flow: nowrap;
  border: 0;
  pointer-events: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 12px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  font-family: 'VTB Group UI',sans-serif;
  box-sizing: border-box;
  border-radius: 4px;
  height: inherit;
  cursor: pointer;
}

.c4 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: 10px;
}

.c4 > * {
  display: block;
  width: 20px;
  height: 20px;
}

.c4 > *:not(:first-child) {
  margin-left: 8px;
}

.c1 {
  cursor: pointer;
  background: #FFFFFF;
  position: relative;
  border-radius: 4px;
  height: 30px;
  border-color: #8A96A8;
  border-width: 1px;
  border-style: solid;
}

[data-focus-within] .c0,
.c1:focus {
  border-color: #0949AA;
}

.c1:hover {
  border-color: #0A54C2;
}

.c1:focus {
  outline: none;
}

.c1:active {
  outline: none;
}

[data-status='success'] .c0 {
  border-color: #2ABB5B;
}

[data-status='error'] .c0 {
  border-color: #DA0B20;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  overflow: hidden;
  position: relative;
  height: 32px;
  padding-right: 46px;
  pointer-events: none;
}

<div>
    <div
      class="c0 c1"
      tabindex="0"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        />
        <div
          class="c4"
        >
          <div
            aria-hidden="true"
            class="c5"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
